lib/src/components/NxDropdown/types.ts

```diff
@@ -4,12 +4,32 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import {ReactNode, HTMLAttributes, ReactElement, WeakValidationMap, KeyboardEventHandler} from 'react';
+import {
+  RefObject,
+  ReactNode, HTMLAttributes, ReactElement, WeakValidationMap, KeyboardEventHandler} from 'react';
 import * as PropTypes from 'prop-types';
 
 import { NX_BUTTON_VARIANTS, NX_BUTTON_VARIANT_TYPE } from '../NxButton/types';
 import { TooltipConfigProps, tooltipPropTypesShape } from '../../util/tooltipUtils';
 
+export interface AbstractDropdownToggleElementProps {
+  onToggleCollapse: (() => void);
+  toggleRef: RefObject<HTMLButtonElement>;
+}
+
+export type AbstractDropdownToggleElement = (props: AbstractDropdownToggleElementProps) => ReactNode;
+
+export type AbstractDropdownProps = Omit<HTMLAttributes<HTMLDivElement>, 'className'> & {
```

Comment:
An `interface` with `extends` is supposed to be a little more performant than an intersection type from what I've heard

Reply:
> An `interface` with `extends` is supposed to be a little more performant than an intersection type from what I've heard

Oo, I didn't know that. Thanks, updated: fcc0e190116829f7cd732cca279e3d6f789f5f69